pkg/reconciler/route/route.go

```diff
@@ -415,6 +417,21 @@ func setTargetsScheme(rs *v1.RouteStatus, dnsNames []string, scheme string) {
 	}
 }
 
+func autoTLSEnabled(ctx context.Context, r *v1.Route) bool {
+	logger := logging.FromContext(ctx)
+	annotationValue := r.Annotations[networking.DisableAutoTLSAnnotationKey]
+	disabledByAnnotation, err := strconv.ParseBool(annotationValue)
+
+	if err != nil {
+		// validation should've caught an invalid value here.
+		// if we have one anyways, assume not disabled and log a warning.
+		logger.Warnf("Invalid annotation value for %q. Value: %q",
+			networking.DisableAutoTLSAnnotationKey, annotationValue)
+	}
+
+	return config.FromContext(ctx).Network.AutoTLS && !disabledByAnnotation
```

Comment:
Should we check this from the beginning? Why do all the parsing etc, if this is _already_ false?

Reply:
yeah good point. i'll restructure this